src/app/Http/Controllers/CrudFeatures/AjaxTable.php

```diff
@@ -8,48 +8,118 @@
-            $row_items = $this->crud->getRowViews($entry, $this->crud);
-
-            // add the buttons as the last column
-            if ($this->crud->buttons->where('stack', 'line')->count()) {
-                $row_items[] = \View::make('crud::inc.button_stack', ['stack' => 'line'])
+        $this->input = $_REQUEST;
+        $this->totalRows = $this->filteredRows = $this->crud->getEntries()->count();
+
+        $data = $this->crud->getEntriesWithConditions(
+            $this->input['length'],
+            $this->input['start'],
+            $this->addAjaxOrderBy()[0],
+            $this->addAjaxOrderBy()[1],
+            $this->addSearchConditions()
+        );
+
+        if ($this->addSearchConditions() !== null) {
+            $this->filteredRows = $data->count();
+        }
+
+        return $this->make($data);
+    }
+
+    /**
+     * Formats the row of the table from the entry(instance of a model).
+     *
+     * @param $entry
+     * @return array
+     */
+    private function format($entry)
+    {
+        $row_items = $this->crud->getRowViews($entry, $this->crud);
+
+        // add the buttons as the last column
+        if ($this->crud->buttons->where('stack', 'line')->count()) {
+            $row_items[] = \View::make('crud::inc.button_stack', ['stack' => 'line'])
                                 ->with('crud', $this->crud)
                                 ->with('entry', $entry)
                                 ->render();
-            }
+        }
 
-            // add the details_row buttons as the first column
-            if ($this->crud->details_row) {
-                array_unshift($row_items, \View::make('crud::columns.details_row_button')
-                                ->with('crud', $this->crud)
-                                ->with('entry', $entry)
-                                ->render());
+        // add the details_row buttons as the first column
+        if ($this->crud->details_row) {
+            array_unshift($row_items, \View::make('crud::columns.details_row_button')
+                                           ->with('crud', $this->crud)
+                                           ->with('entry', $entry)
+                                           ->render());
+        }
+
+        return $row_items;
+    }
+
+    /**
+     * Created the array to be fed to the data table.
+     * @param $data
+     * @return array
+     */
+    private function make($data)
+    {
+        $rows = [];
+        foreach ($data as $row) {
+            $rows[] = $this->format($row);
+        }
+
+        return [
+            'draw'            => (isset($this->input['draw']) ? (int) $this->input['draw'] : 0),
+            'recordsTotal'    => $this->totalRows,
+            'recordsFiltered' => $this->filteredRows,
+            'data'            => $rows,
+        ];
+    }
+
+    /**
+     * Checks of teh user has anything written in the search bar and if so a string with the filtered is returned.
```

Comment:
If my browser keeps on disallowing me from writing **t** **e** **h** I'm going to scream :P Check the spelling :P